lib/pbench/server/api/resources/query_apis/datasets/datasets_contents.py

```diff
@@ -65,14 +66,7 @@ def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
 
         # Retrieve the ES indices that belong to this run_id from the metadata
         # table
-
         indices = self.get_index(dataset, "run-toc")
-        if not indices:
-            self.logger.debug(
-                f"Found no indices matching the prefix run-toc"
-                f" for a dataset {dataset!r}"
-            )
-            abort(HTTPStatus.NOT_FOUND, message="Found no matching indices")
 
         return {
             "path": f"/{indices}/_search",
```

Comment:
Why don't we need to protect against an empty return from `get_index()`?

Reply:
Because `get_index` will throw `APIAbort` in that case, and we just let it propagate out to `ApiBase._dispatch`.